src/object_store/src/object/mod.rs

```diff
@@ -651,130 +669,158 @@ impl<OS: ObjectStore> MonitoredObjectStore<OS> {
         path: &str,
         range: Range<usize>,
     ) -> ObjectResult<MonitoredStreamingReader> {
-        let operation_type = "streaming_read_start";
+        let operation_type = OperationType::StreamingReadInit;
+        let operation_type_str = operation_type.as_str();
         let media_type = self.media_type();
         let _timer = self
             .object_store_metrics
             .operation_latency
-            .with_label_values(&[media_type, operation_type])
+            .with_label_values(&[media_type, operation_type_str])
             .start_timer();
-        let future = self
-            .inner
-            .streaming_read(path, range)
-            .verbose_instrument_await("object_store_streaming_read");
-        let res = match self.streaming_read_timeout.as_ref() {
-            None => future.await,
-            Some(timeout) => tokio::time::timeout(*timeout, future)
+
+        let retry_condition = RetryCondition::new(
+            self.config.retry.streaming_read_timeout_ms,
+            operation_type,
+            self.object_store_metrics.clone(),
+        );
+
+        let builder = || async {
+            self.inner
+                .streaming_read(path, range.clone())
+                .verbose_instrument_await(operation_type_str)
                 .await
-                .unwrap_or_else(|_| Err(ObjectError::internal("streaming_read init timeout"))),
         };
 
-        try_update_failure_metric(&self.object_store_metrics, &res, operation_type);
+        let res = retry_request(builder, &self.config, operation_type, retry_condition).await;
```

Comment:
ditto. Will it be clearer to move the init timeout logic under the specific streaming reader implementaion given that the read bytes timeout is done inside the implementation. In this case, all streaming reader related timeout is maintained inside the specific implementaion.